<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainUi</class>
 <widget class="QMainWindow" name="MainUi">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1728</width>
    <height>1051</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>ImageOperationModel</string>
  </property>
  <property name="styleSheet">
   <string notr="true">

                QWidget {
                background:qlineargradient(spread:pad,x1:0,y1:0,x2:0,y2:1,stop:0 #708090,stop:1 #C8C8D0);
                }

                QMenuBar {
                min-width: 80px;
                min-height: 35px;
                }

                QMenu {
                border-radius: 30px;
                }
                QMenu::item {
                min-width: 60px;
                min-height: 40px;
                padding: 2px 25px 2px 30px;
                margin-left: 5px;
                }
                QMenu::item:selected {
                border-width:1px;
                border-color: #516589;
                background: qlineargradient(x1: 0, y1: 0, x2: 0, y2: 1, stop: 0 rgb(45,133,207), stop: 1.0
                rgb(125,195,250));
                color:#E6FFFF;
                }

                QPushButton {
                min-width:100px;
                min-height:40px;
                border-radius:10px;
                }

                QPushButton:hover
                {
                background-color:rgb(44 , 137 , 255);
                }

                QPushButton:pressed
                {
                background-color:rgb(14 , 135 , 228);
                padding-left:3px;
                padding-top:3px;
                }

                QScrollBar:horizontal {
                border: 1px solid #222222;
                background: QLinearGradient( x1: 0, y1: 0, x2: 0, y2: 1, stop: 0.0 #ffffff, stop: 0.5 #ffffff, stop: 1
                #ffffff);
                height: 7px;
                margin: 0px 16px 0 16px;
                }
                QScrollBar::handle:horizontal {
                background: QLinearGradient( x1: 0, y1: 0, x2: 1, y2: 0, stop: 0 #b1b1b1, stop: 0.5 #b3b3b3, stop: 1
                #b1b1b1);
                min-height: 20px;
                border-radius: 20px;
                }
                QScrollBar::add-line:horizontal {
                border: 1px solid #1b1b19;
                border-radius: 20px;
                background: QLinearGradient( x1: 0, y1: 0, x2: 1, y2: 0, stop: 0 #999999, stop: 1 #999999);
                width: 14px;
                subcontrol-position: right;
                subcontrol-origin: margin;
                }
                QScrollBar::sub-line:horizontal {
                border: 1px solid #1b1b19;
                border-radius: 20px;
                background: QLinearGradient( x1: 0, y1: 0, x2: 1, y2: 0, stop: 0 #999999, stop: 1 #999999);
                width: 14px;
                subcontrol-position: left;
                subcontrol-origin: margin;
                }
                QScrollBar::right-arrow:horizontal, QScrollBar::left-arrow:horizontal {
                border: 1px solid black;
                width: 1px;
                height: 1px;
                background: white;
                }
                QScrollBar::add-page:horizontal, QScrollBar::sub-page:horizontal {
                background: none;
                }
                QScrollBar:vertical {
                background: QLinearGradient( x1: 0, y1: 0, x2: 1, y2: 0, stop: 0.0 #ffffff, stop: 0.5 #ffffff, stop: 1
                #ffffff);
                width: 7px;
                margin: 16px 0 16px 0;
                border: 1px solid #222222;
                }
                QScrollBar::handle:vertical {
                background: QLinearGradient( x1: 0, y1: 0, x2: 0, y2: 1, stop: 0 #b1b1b1, stop: 0.5 #b3b3b3, stop: 1
                #b1b1b1);
                min-height: 20px;
                border-radius: 20px;
                }
                QScrollBar::add-line:vertical {
                border: 1px solid #1b1b19;
                border-radius: 20px;
                background: QLinearGradient( x1: 0, y1: 0, x2: 0, y2: 1, stop: 0 #999999, stop: 1 #999999);
                height: 14px;
                subcontrol-position: bottom;
                subcontrol-origin: margin;
                }
                QScrollBar::sub-line:vertical {
                border: 1px solid #1b1b19;
                border-radius: 20px;
                background: QLinearGradient( x1: 0, y1: 0, x2: 0, y2: 1, stop: 0 #999999, stop: 1 #999999);
                height: 14px;
                subcontrol-position: top;
                subcontrol-origin: margin;
                }
                QScrollBar::up-arrow:vertical, QScrollBar::down-arrow:vertical {
                border: 1px solid black;
                width: 1px;
                height: 1px;
                background: white;
                }
                QScrollBar::add-page:vertical, QScrollBar::sub-page:vertical {
                background: none;
                }

                QTabWidget::pane {
                border: 1px solid #444;
                top: 1px;
                }

                QTabBar::tab {
                color: #ffffff;
                border: 1px solid #444;
                border-bottom-style: none;
                background-color: #999999;
                padding-left: 10px;
                padding-right: 10px;
                padding-top: 3px;
                padding-bottom: 2px;
                margin-right: -1px;
                min-width: 80px;
                min-height: 30px;
                }
                QTabBar::tab:last {
                margin-right: 0;
                border-top-right-radius: 3px;
                }
                QTabBar::tab:first:!selected {
                margin-left: 0px;
                border-top-left-radius: 3px;
                }
                QTabBar::tab:!selected {
                border-bottom-style: solid;
                margin-top: 3px;
                background-color: QLinearGradient(x1:0, y1:0, x2:0, y2:1, stop:1 #1874CD, stop:.4 #737373);
                }
                QTabBar::tab:selected {
                border-top-left-radius: 3px;
                border-top-right-radius: 3px;
                margin-bottom: 0px;
                }
                QTabBar::tab:!selected:hover {
                border-top-left-radius: 3px;
                border-top-right-radius: 3px;
                background-color: QLinearGradient(x1:0, y1:0, x2:0, y2:1, stop:1 #212121, stop:0.4 #737373, stop:0.2
                #999999, stop:0.1 #b1b1b1);
                }

            </string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="sizePolicy">
    <sizepolicy hsizetype="Maximum" vsizetype="Maximum">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="layoutDirection">
    <enum>Qt::LeftToRight</enum>
   </property>
   <property name="autoFillBackground">
    <bool>false</bool>
   </property>
   <property name="inputMethodHints">
    <set>Qt::ImhNone</set>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout_5">
    <item>
     <layout class="QVBoxLayout" name="openImageVerticalLayout">
      <item>
       <layout class="QHBoxLayout" name="openImageButtonHorizontalLayout">
        <item>
         <widget class="QPushButton" name="preImageButton">
          <property name="text">
           <string>上一张</string>
          </property>
          <property name="autoRepeat">
           <bool>false</bool>
          </property>
          <property name="autoDefault">
           <bool>false</bool>
          </property>
          <property name="flat">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="openImageHorizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QPushButton" name="nextImageButton">
          <property name="text">
           <string>下一张</string>
          </property>
          <property name="flat">
           <bool>false</bool>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QScrollArea" name="openImageFileScroll">
        <property name="layoutDirection">
         <enum>Qt::LeftToRight</enum>
        </property>
        <property name="verticalScrollBarPolicy">
         <enum>Qt::ScrollBarAsNeeded</enum>
        </property>
        <property name="horizontalScrollBarPolicy">
         <enum>Qt::ScrollBarAsNeeded</enum>
        </property>
        <property name="widgetResizable">
         <bool>true</bool>
        </property>
        <property name="alignment">
         <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
        </property>
        <widget class="QWidget" name="openImageFileScrollWidget">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>0</y>
           <width>840</width>
           <height>894</height>
          </rect>
         </property>
         <property name="layoutDirection">
          <enum>Qt::LeftToRight</enum>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout_4">
          <item>
           <widget class="QLabel" name="openImageFileLabel">
            <property name="layoutDirection">
             <enum>Qt::LeftToRight</enum>
            </property>
            <property name="text">
             <string/>
            </property>
            <property name="alignment">
             <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tabWidget1">
       <attribute name="title">
        <string>Tab1</string>
       </attribute>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <widget class="QScrollArea" name="tabScrollArea1">
          <property name="widgetResizable">
           <bool>true</bool>
          </property>
          <widget class="QWidget" name="tabInnerWidget1">
           <property name="geometry">
            <rect>
             <x>0</x>
             <y>0</y>
             <width>800</width>
             <height>879</height>
            </rect>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout">
            <item>
             <widget class="QLabel" name="tabLabel1">
              <property name="text">
               <string/>
              </property>
              <property name="alignment">
               <set>Qt::AlignCenter</set>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tabWidget2">
       <attribute name="title">
        <string>Tab2</string>
       </attribute>
       <layout class="QHBoxLayout" name="horizontalLayout_3">
        <item>
         <widget class="QScrollArea" name="tabScrollArea2">
          <property name="widgetResizable">
           <bool>true</bool>
          </property>
          <widget class="QWidget" name="tabInnerWidget2">
           <property name="geometry">
            <rect>
             <x>0</x>
             <y>0</y>
             <width>83</width>
             <height>40</height>
            </rect>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_2">
            <item>
             <widget class="QLabel" name="tabLabel2">
              <property name="text">
               <string/>
              </property>
              <property name="alignment">
               <set>Qt::AlignCenter</set>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tabWidget3">
       <property name="focusPolicy">
        <enum>Qt::NoFocus</enum>
       </property>
       <attribute name="title">
        <string>Tab3</string>
       </attribute>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="QScrollArea" name="scrollArea">
          <property name="widgetResizable">
           <bool>true</bool>
          </property>
          <widget class="QWidget" name="tabInnerWidget3">
           <property name="geometry">
            <rect>
             <x>0</x>
             <y>0</y>
             <width>83</width>
             <height>40</height>
            </rect>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_3">
            <item>
             <widget class="QLabel" name="tabLabel3">
              <property name="text">
               <string/>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tabWidget4">
       <attribute name="title">
        <string>Tab4</string>
       </attribute>
       <layout class="QHBoxLayout" name="horizontalLayout_6">
        <item>
         <widget class="QScrollArea" name="scrollArea_2">
          <property name="widgetResizable">
           <bool>true</bool>
          </property>
          <widget class="QWidget" name="tabInnerWidget4">
           <property name="geometry">
            <rect>
             <x>0</x>
             <y>0</y>
             <width>83</width>
             <height>40</height>
            </rect>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_4">
            <item>
             <widget class="QLabel" name="tabLabel4">
              <property name="text">
               <string/>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tabWidget5">
       <attribute name="title">
        <string>Tab5</string>
       </attribute>
       <layout class="QHBoxLayout" name="horizontalLayout_7">
        <item>
         <widget class="QScrollArea" name="scrollArea_3">
          <property name="widgetResizable">
           <bool>true</bool>
          </property>
          <widget class="QWidget" name="tabInnerWidget5">
           <property name="geometry">
            <rect>
             <x>0</x>
             <y>0</y>
             <width>83</width>
             <height>40</height>
            </rect>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_5">
            <item>
             <widget class="QLabel" name="label_3">
              <property name="text">
               <string/>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1728</width>
     <height>37</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>文件</string>
    </property>
    <addaction name="actionOpen"/>
    <addaction name="actionSave"/>
    <addaction name="actionOpenFilePath"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuOpenCV">
    <property name="title">
     <string>OpenCV示例</string>
    </property>
    <widget class="QMenu" name="menuGeometricTransformation">
     <property name="title">
      <string>几何变换</string>
     </property>
     <addaction name="actionResize"/>
     <addaction name="actionTranslation"/>
     <addaction name="actionRotation"/>
     <addaction name="actionWarpAffine"/>
     <addaction name="actionWarpPerspective"/>
    </widget>
    <widget class="QMenu" name="menu">
     <property name="title">
      <string>图像阈值</string>
     </property>
     <addaction name="actionSimpleBinaryThreshold"/>
     <addaction name="actionSimpleBinaryInvThreshold"/>
     <addaction name="actionSimpleTruncThreshold"/>
     <addaction name="actionSimpleToZeroThreshold"/>
     <addaction name="actionSimpleToZeroInvThreshold"/>
     <addaction name="actionAdaptiveMeanThreshold"/>
     <addaction name="actionAdaptiveGaussianThreshold"/>
     <addaction name="actionOTSUThreshold"/>
    </widget>
    <addaction name="menuGeometricTransformation"/>
    <addaction name="menu"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuOpenCV"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionOpen">
   <property name="text">
    <string>打开图片</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="checkable">
    <bool>false</bool>
   </property>
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>保存</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>退出</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionOpenFilePath">
   <property name="text">
    <string>读取文件夹</string>
   </property>
  </action>
  <action name="actionResize">
   <property name="text">
    <string>缩放</string>
   </property>
  </action>
  <action name="actionTranslation">
   <property name="text">
    <string>平移</string>
   </property>
  </action>
  <action name="actionRotation">
   <property name="text">
    <string>旋转</string>
   </property>
  </action>
  <action name="actionWarpAffine">
   <property name="text">
    <string>仿射变换</string>
   </property>
  </action>
  <action name="actionWarpPerspective">
   <property name="text">
    <string>透视变换</string>
   </property>
  </action>
  <action name="actionSimpleBinaryThreshold">
   <property name="text">
    <string>简单二值化阈值</string>
   </property>
  </action>
  <action name="actionSimpleBinaryInvThreshold">
   <property name="text">
    <string>简单反二值化阈值</string>
   </property>
  </action>
  <action name="actionSimpleTruncThreshold">
   <property name="text">
    <string>简单截断阈值</string>
   </property>
  </action>
  <action name="actionSimpleToZeroThreshold">
   <property name="text">
    <string>简单置零阈值</string>
   </property>
  </action>
  <action name="actionSimpleToZeroInvThreshold">
   <property name="text">
    <string>简单置零反阈值</string>
   </property>
  </action>
  <action name="actionAdaptiveMeanThreshold">
   <property name="text">
    <string>自适应均值阈值</string>
   </property>
  </action>
  <action name="actionAdaptiveGaussianThreshold">
   <property name="text">
    <string>自适应高斯阈值</string>
   </property>
  </action>
  <action name="actionOTSUThreshold">
   <property name="text">
    <string>OTSU二值化</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
